ui-src/src/drawing/layoutFormula.js

```diff
@@ -65,5 +68,10 @@ export default function layoutFormula(screenWidth, goals, edges) {
   const withHierarchies = goals.map(g => mapGoalToHierarchy(g, edges))
 
   // use positions in the hierarchy to determine coordinates
-  return withHierarchies.map(wH => mapHierarchyToPosition(wH, withHierarchies, screenWidth))
+  const coordinates = {}
```

Comment:
I refactored the results of 'layoutFormula' to return an object with the coordinates keyed by the Goal address, instead of just an array, where the addresses were implicitly mapped to the addresses in an array of addresses. 

This makes the coordinates more accessible by consumers of this function, because they can get the coordinates of the goal without knowing the index in the array, simply knowing the address.